Meter.h

```diff
@@ -20,13 +20,37 @@ in the source distribution for its full text.
 
 #define METER_BUFFER_LEN 256
 
+#define METER_BUFFER_CHECK(buffer, size, written)          \
+   do {                                                    \
+      if ((written) < 0 || (size_t)(written) >= (size)) {  \
+         return;                                           \
+      }                                                    \
+      buffer += written;                                   \
+      size -= (size_t)written;                             \
+   } while (0)
+
+#define METER_BUFFER_APPEND_CHR(buffer, size, c)   \
+   do {                                            \
+      if ((size) == 0) {                           \
+         return;                                   \
+      }                                            \
+      *buffer = c;                                 \
+      buffer += 1;                                 \
+      *buffer = '\0';                              \
+      size--;                                      \
+      if ((size) == 0) {                           \
+         return;                                   \
+      }                                            \
+   } while (0)
```

Comment:
Still allows for buffer overrun as it is. What about this?

```suggestion
#define METER_BUFFER_APPEND_CHR(buffer, size, c)   \
   do {                                            \
      if ((size) < 2) {                            \
         return;                                   \
      }                                            \
      *buffer++ = c;                               \
      size--;                                      \
      *buffer = '\0';                              \
      if ((size) <= 1) {                           \
         return;                                   \
      }                                            \
   } while (0)
```

Reply:
Also: Maybe align the final backslashes with the previous macro?